src/services/RetroGenerationService.ts

```diff
@@ -44,31 +44,35 @@ class RetroGenerationService {
                     const detailsTemplate: any = template.reportTemplate.activityDetails[i];
                     const testResult: any = testResults[j];
                     const testType: any = testResult.testTypes;
-                    const additionalTestTypeNotes = testType.prohibitionIssued ? 'Prohibition was issued' : 'Prohibition was not issued'
+                    const additionalTestTypeNotes: string = testType.prohibitionIssued ? 'Prohibition was issued' : 'Prohibition was not issued';
+                    let defects: string = "";
+                    let reasonForAbandoning: string = "";
+                    let additionalCommentsAbandon: string = "";
+                    let defectsDetails: string = "";
+                    let prsString: string = "";
 
-                    //TODO Refactor
-                    let prsString = 'PRS';
-                    try { if (testType.defects[i].prs === undefined) prsString = 'PRS';} catch (e) {
-                        prsString = 'NOT PRS';
-                    }
+                    for(let i = 0; i < testType.defects.length; i++) {
+                        testType.defects[i].prs ? prsString = ", PRS" : "";
```

Comment:
![image](https://user-images.githubusercontent.com/44023301/57306739-07fb1900-70ec-11e9-9943-1ea90f1d11c8.png)
This is a bug in sonarqube. please fix it

Reply:
Also please run the linter. There are a lot of errors

Reply:
Linter issues are now fixed. Could you check the sonarqube bug again? I have changed the code so it should work now.